frontend/src/metabase/query_builder/components/NativeQueryEditor/CodeMirrorEditor/completers.tsx

```diff
@@ -73,15 +74,10 @@ export function useSchemaCompletion({ databaseId }: SchemaCompletionOptions) {
 
       const suffix = matchAfter(context, /\w+/);
 
-      // Do not cache this in the rtk-query's cache, since there is no
-      // way to invalidate it. The autocomplete endpoint set caching headers
-      // so the request should be cached in the browser.
-      const shouldCache = false;
-
-      const { data } = await getAutocompleteSuggestions(
-        { databaseId, matchStyle, query: word.text.trim() },
-        shouldCache,
-      );
+      const { data } = await listAutocompleteSuggestions({
+        databaseId,
+        [matchStyle]: word.text.trim(),
+      });
```

Comment:
Is there a reason you removed the `shouldCache` argument here?

This PR adds tags to the requests so they can be invalidated, but AFAICT that does not handle the case where the server rescans (or is there some mechanism that informs all clients whan that happens?)

Reply:
Yes, when you "rescan" we invalidate the RTK cache.